### DISCUSSION TITLE
Login page for API route /api/v1/transactions when using POST

### DISCUSSION THREAD

**Discussion by xEtarusx** [firefly-iii/firefly-iii#6235] [Firefly III] [answered]
Hi there,
I'm currently experimenting with the Firefly API and trying to insert new transactions. But currently I only get the login page as a response when I use Postman with a personal token.

Route: POST <baseUrl>/api/v1/transactions
Header: "Authorization: Bearer \<token>"
Body: 
```json
{
	"error_if_duplicate_hash": false,
	"apply_rules": false,
	"fire_webhook": false,
	"group_title": "",
	"transactions": [
		{
			"type": "withdrawal",
			"date": "2022-07-16T00:00:00Z",
			"amount": "-66,00",
			"category_name": "Gas",
			"source_name": "Girokonto",
			"destination_name": "Stadtwerke",
			"notes": ""
		},
		{
			"type": "withdrawal",
			"date": "2022-07-16T00:00:00Z",
			"amount": "-80,00",
			"category_name": "Strom",
			"source_name": "Girokonto",
			"destination_name": "Stadtwerke",
			"notes": ""
		}
	]
}
```

But I only get a 200 response with the Firefly login page as response body.
The docker log shows the following entries on a POST request:
```
fireflyiii                | 192.168.188.20 - - [17/Jul/2022:09:21:28 +0200] "POST /api/v1/transactions HTTP/1.0" 302 1362 "-" "PostmanRuntime/7.29.0"
fireflyiii                | 192.168.188.20 - - [17/Jul/2022:09:21:28 +0200] "GET / HTTP/1.0" 302 2291 "<baseUrl>/api/v1/transactions" "PostmanRuntime/7.29.0"
fireflyiii                | [2022-07-17T09:21:28.479952+02:00] local.INFO: AUDIT: Show login form (1.1). (192.168.188.28 -> GET:<baseUrl>/login) [] []
fireflyiii                | 192.168.188.20 - - [17/Jul/2022:09:21:28 +0200] "GET /login HTTP/1.0" 200 3488 "<baseUrl>" "PostmanRuntime/7.29.0"
```

However when I switch from POST to GET I can retrieve all current transactions from Firefly with no problems at all. 
So I'm confused that the token is just not working when using the POST request? Am I missing something here?

Any help is appreciated :)
Thank you

**Comment by JC5** [comment, marked as answer]
Make sure you add the `Accept: application/json` header

**Reply by xEtarusx** [reply]
Okay that did the trick.
I already had the accept header in the request with  `Accept: */*`. 
So it looks like wildcards are not accepted, not sure if this is a bug or on purpose.
But I will stick to explicit application/json for now.

Thank you :)

**Reply by JC5** [reply]
It's on purpose.
